tt_eager/tt_dnn/op_library/bmm/multi_core_reuse_mcast_dram_sharded_optimized/bmm_op_multi_core_reuse_dram_sharded_optimized.cpp

```diff
@@ -1146,26 +1152,22 @@ operation::ProgramWithCallbacks matmul_multi_core_reuse_dram_sharded_optimized_(
     uint32_t in1_single_tile_size = tt_metal::detail::TileSize(in1_data_format);
     tt_metal::Buffer* in0_buffer = a.buffer();
     tt_metal::Buffer* in1_buffer = b.buffer();
-    if (bcast_batch)
-        TT_FATAL(
-            bshape[0] * bshape[1] == 1 &&
-            "matmul (batch bcast variant) expects input tensors of shapes BCMK*11KN=BCMN");
-    else {
-        // same condition as above, different message
+    TT_FATAL(ashape.rank() == bshape.rank() && "bmm (non-bcast matmul) expects input tensors of the same rank");
```

Comment:
Do we need to assert for rank 2?

Reply:
ah wdym, assert for rank >= 2 or sth else?

Reply:
yeah my bad, assert for at least rank two

Reply:
added